seed_embeddings/OpenKE/generate_embedding_ray.py

```diff
@@ -221,81 +225,56 @@ def findRep(src, dest, index_dir, src_type="json"):
         )
-        tuner = tune.Tuner(
-            train,
-            param_space=search_space,
-            tune_config=TuneConfig(
-                max_concurrent_trials=4,
-                scheduler=scheduler,
-                num_samples=1,
-            ),
-            run_config=RunConfig(
-                checkpoint_config=CheckpointConfig(
-                    num_to_keep=2,
-                    # *Best* checkpoints are determined by these params:
-                    # checkpoint_score_attribute="AnalogiesScore",
-                    # checkpoint_score_order="max",
-                )
-            ),
-        )
     elif arg_conf.link_pred:
         scheduler = ASHAScheduler(
             time_attr="training_iteration",
             max_t=arg_conf.epoch,
-            grace_period=min(arg_conf.epoch, 4000),
-            reduction_factor=2,
+            grace_period=15,
+            reduction_factor=3,
             metric="hit1",
             mode="max",
         )
-        tuner = tune.Tuner(
-            train,
-            param_space=search_space,
-            tune_config=TuneConfig(
-                max_concurrent_trials=4,
-                scheduler=scheduler,
-                num_samples=1,
-            ),
-            run_config=RunConfig(
-                checkpoint_config=CheckpointConfig(
-                    num_to_keep=2,
-                    # *Best* checkpoints are determined by these params:
-                    # checkpoint_score_attribute="hit1",
-                    # checkpoint_score_order="max",
-                )
-            ),
-        )
     else:
         scheduler = ASHAScheduler(
             time_attr="training_iteration",
             max_t=arg_conf.epoch,
-            grace_period=min(arg_conf.epoch, 4000),
-            reduction_factor=2,
+            grace_period=15,
+            reduction_factor=3,
             metric="loss",
             mode="min",
         )
-        tuner = tune.Tuner(
-            train,
-            param_space=search_space,
-            tune_config=TuneConfig(
-                max_concurrent_trials=4,
-                scheduler=scheduler,
-                num_samples=1,
-            ),
-            run_config=RunConfig(
-                checkpoint_config=CheckpointConfig(
-                    num_to_keep=2,
-                    # *Best* checkpoints are determined by these params:
-                    # checkpoint_score_attribute="loss",
-                    # checkpoint_score_order="min",
-                )
-            ),
+    if arg_conf.use_gpu:
+        train_with_resources = tune.with_resources(
+            train, resources={"cpu": 0, "gpu": 0.0625}
+        )
+    else:
+        train_with_resources = tune.with_resources(
+            train, resources={"cpu": 10, "gpu": 0}
         )
 
+    tuner = tune.Tuner(
+        train_with_resources,
+        param_space=search_space,
+        tune_config=TuneConfig(
+            search_alg=OptunaSearch(metric="loss", mode="min"),
```

Comment:
This optimization/search depends on the objective. Please create an object with appropriate objective in the if/else cases and use that object here.